tests/test_posixDatastore.py

```diff
@@ -25,77 +25,92 @@
-        self.assertEqual(inputRecord.getPsfFluxFlag(), outputRecord.getPsfFluxFlag())
-        self.assertEqual(inputTable.getCentroidDefinition(), outputTable.getCentroidDefinition())
-        self.assertEqual(inputRecord.getCentroid(), outputRecord.getCentroid())
-        self.assertFloatsAlmostEqual(
-            inputRecord.getCentroidErr()[0, 0],
-            outputRecord.getCentroidErr()[0, 0], rtol=1e-6)
-        self.assertFloatsAlmostEqual(
-            inputRecord.getCentroidErr()[1, 1],
-            outputRecord.getCentroidErr()[1, 1], rtol=1e-6)
-        self.assertEqual(inputTable.getShapeDefinition(), outputTable.getShapeDefinition())
-        self.assertFloatsAlmostEqual(
-            inputRecord.getShapeErr()[0, 0],
-            outputRecord.getShapeErr()[0, 0], rtol=1e-6)
-        self.assertFloatsAlmostEqual(
-            inputRecord.getShapeErr()[1, 1],
-            outputRecord.getShapeErr()[1, 1], rtol=1e-6)
-        self.assertFloatsAlmostEqual(
-            inputRecord.getShapeErr()[2, 2],
-            outputRecord.getShapeErr()[2, 2], rtol=1e-6)
-
     def testConstructor(self):
         datastore = PosixDatastore(config=self.configFile)
         self.assertIsNotNone(datastore)
 
     def testBasicPutGet(self):
-        catalog = datasetsHelper.makeExampleCatalog()
+        metrics = makeExampleMetrics()
         datastore = PosixDatastore(config=self.configFile)
         # Put
-        storageClass = datastore.storageClassFactory.getStorageClass("SourceCatalog")
-        uri, _ = datastore.put(catalog, storageClass=storageClass, storageHint="tester.fits", typeName=None)
+        storageClass = datastore.storageClassFactory.getStorageClass("StructuredData")
+        uri, comps = datastore.put(metrics, storageClass=storageClass, storageHint="tester.json",
+                                   typeName=None)
+
         # Get
-        catalogOut = datastore.get(uri, storageClass=storageClass, parameters=None)
-        datasetsHelper.assertCatalogEqual(self, catalog, catalogOut)
+        metricsOut = datastore.get(uri, storageClass=storageClass, parameters=None)
+        self.assertEqualMetrics(metrics, metricsOut)
+
+        # Get a component even though it was written without
+        summary = datastore.get("{}#{}".format(uri, "summary"), storageClass=storageClass)
+        self.assertEqual(summary, metricsOut.summary)
+
         # These should raise
         with self.assertRaises(ValueError):
             # non-existing file
-            datastore.get(uri="file:///non_existing.fits", storageClass=storageClass, parameters=None)
+            datastore.get(uri="file:///non_existing.json", storageClass=storageClass, parameters=None)
         with self.assertRaises(ValueError):
             # invalid storage class
-            datastore.get(uri="file:///non_existing.fits", storageClass=object, parameters=None)
+            datastore.get(uri="file:///non_existing.json", storageClass=object, parameters=None)
```

Comment:
We should be careful that we don't start relying on file extensions anywhere (e.g. object stores don't have them).

Reply:
I'm not looking at file extensions anywhere at the moment. That get will fail regardless.

Reply:
I do now support file extensions inside formatters because some I/O libraries get upset if you don't have `.fits` (for example). The hint is treated as a hint and you get back a URI with the right extension.